onadata/apps/api/permissions.py

```diff
@@ -170,6 +170,20 @@ def has_permission(self, request, view):
         is_authenticated = request and request.user.is_authenticated()
 
         if is_authenticated and view.action == 'create':
+
+            # Handle bulk create
+            # if doing a bulk create we will fail the entire process if the
+            # user lacks permissions for even one instance
+            if isinstance(request.data, list):
+                instance_ids = list(set([_['instance'] for _ in request.data]))
+                xforms = XForm.objects.filter(
+                    instances__in=instance_ids).distinct()
```

Comment:
Could we avoid the join query and qury instance model directly? I suspect `has_perm()` is only going to require the `xform_id` and not the whole xform with XML + JSON data.

```python
submissions = Instance.objects.filter(id__in=instance_ids).only('xform').distinct():
for submission in submissions:
    if not request.user.has_perm(CAN_CHANGE_XFORM, xform):
...
```